### PR TITLE
soc: nxp_imx: Add support for TRNG

### PR DESCRIPTION
**Pull request by aurel32**
Add support for the TRNG device contained in the i.MX RT SoCs. It uses
the existing MCUX driver, and mostly consists in adding the Kconfig and
DTS entries.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

### CODE REVIEW

File: drivers/entropy/entropy_mcux_trng.c [hunk 1]
```diff
@@ -11,6 +11,14 @@
...
 #include "fsl_trng.h"
 
+/* The True Random Number Generator device is called "TRNG0" on the
+ * Kinetis SoCs and "TRNG" on the I.MX RT SoC.  Add a define to cope
+ * with that.
+ */
+#ifndef TRNG0
+#define TRNG0 TRNG
+#endif
+
```
Comment (jhqian) [inline drivers/entropy/entropy_mcux_trng.c]: I don't think it's a good idea. After checking the code, I would suggest not to hard coding TRNG0 in this file, but to use something like  TRNG_BASE_PTRS[0] as a workaround, given the different macro of this base address.

**Review by jhqian — CHANGES_REQUESTED** [review]

File: drivers/entropy/entropy_mcux_trng.c (hunk 1 above)
Comment (aurel32) [reply drivers/entropy/entropy_mcux_trng.c]: Thanks for your suggestion, I have just pushed a new version fixing that.

Comment (MaureenHelm) [reply drivers/entropy/entropy_mcux_trng.c]: Actually you should get the base pointer from device tree instead. Please make a separate commit to change the driver to use `DT_ENTROPY_MCUX_TRNG_BASE_ADDRESS` below, then a second commit to enable the driver on rt.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/entropy/entropy_mcux_trng.c (hunk 1 above)
Comment (aurel32) [reply drivers/entropy/entropy_mcux_trng.c]: I didn't realize that `TRNG` and `TRNG0` corresponded to the base address of the peripheral, thanks for the hint. I have just reworked the PR as you requested.

**Review by MaureenHelm — APPROVED** [review]

**Review by jhqian — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**